### PR TITLE
Migrated TC test_self_heal_with_link_files.py

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
[Migrated TC test_self_heal_with_link_files.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/afr/heal/test_self_heal_with_link_files.py)
    
Updates: #292
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

### CODE REVIEW

**Comment by Sheetalpamecha** [comment]
```
Results for heal
Test nature : disruptive
TestName : test_self_heal_with_link_files
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|   dist-rep  |     PASS    |        00:05:37       |      NA     |
|     rep     |     PASS    |        00:04:55       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   1   |
|         Non Disruptive Tests         |   0   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   2   |
|            Tests Skipped             |   0   |
|             Total Tests              |   1   |
|   Disruptive Tests Pass Percentage   | 100.0 |
| Non Disruptive Tests Pass Percentage |   0   |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 11 minutes 02 seconds
✔ Tearing down successful.

```

File: tests/functional/afr/heal/test_self_heal_with_link_files.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        if not second_attempt:
+            self.redant.create_dir(self.mountpoint,
+                                   'test_link_self_heal',
+                                   self.client_list[0])
+
+        # Create dirctories and files inside directory test_link_self_heal
+        io_cmd = (f"cd {self.mountpoint}/test_link_self_heal;"
+                  "for i in `seq 1 5`; do mkdir dir.$i; "
+                  "for j in `seq 1 10`; do dd if=/dev/random "
+                  "of=dir.$i/file.$j bs=1k count=$j; done; done")
```
Comment (nik-redhat) [inline tests/functional/afr/heal/test_self_heal_with_link_files.py:37]: I think can go under the `if` part ?

File: tests/functional/afr/heal/test_self_heal_with_link_files.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        if second_attempt:
```
Comment (nik-redhat) [inline tests/functional/afr/heal/test_self_heal_with_link_files.py:38]: And, after making the above change, we can simply use an else here.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: tests/functional/afr/heal/test_self_heal_with_link_files.py (hunk 1 above)
Comment (Sheetalpamecha) [reply tests/functional/afr/heal/test_self_heal_with_link_files.py]: Yes

File: tests/functional/afr/heal/test_self_heal_with_link_files.py (hunk 2 above)
Comment (Sheetalpamecha) [reply tests/functional/afr/heal/test_self_heal_with_link_files.py]: Done

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**